micov/_io.py

```diff
@@ -190,6 +209,38 @@ def parse_genome_lengths(lengths):
     return df[[genome_id_col, length_col]].rename(rename)
 
 
+def parse_taxonomy(taxonomy):
+    with open(taxonomy) as fp:
+        first_line = fp.readline()
+
+    has_header = _test_has_header_taxonomy(first_line)
+    df = pl.read_csv(taxonomy, separator='\t', has_header=has_header)
+    genome_id_col = df.columns[0]
+    taxonomy_col = df.columns[1]
+
+    genome_ids = df[genome_id_col]
+    if len(genome_ids) != len(set(genome_ids)):
+        raise ValueError(f"'{genome_id_col}' is not unique")
+    
+    rename = {genome_id_col: COLUMN_GENOME_ID,
+              taxonomy_col: COLUMN_TAXONOMY}
+    
+    return df[[genome_id_col, taxonomy_col]].rename(rename)
+
+
+def set_taxonomy_as_id(coverages, taxonomy):
+    missing = (set(coverages[COLUMN_GENOME_ID]) -
+               set(taxonomy[COLUMN_GENOME_ID]))
+    if len(missing) > 0:
+        raise ValueError(f"{len(missing)} genome(s) appear unrepresented in "
+                         f"the taxonomy information, examples: "
+                         f"{sorted(missing)[:5]}")
+
+    coverages_with_taxonomy = coverages.join(taxonomy, on=COLUMN_GENOME_ID, how='left')
+    coverages_with_taxonomy = coverages_with_taxonomy.select(COLUMN_TAXONOMY, pl.exclude(COLUMN_GENOME_ID, COLUMN_TAXONOMY))
+    return coverages_with_taxonomy
```

Comment:
If this isn't already lazy coming in, then the following would work. If it is lazy, then the `.lazy()` and `.collect()` calls can be omitted

Is it correct to select `COLUMN_TAXONOMY` and also exclude `COLUMN_TAXONOMY`? 

```suggestion
    return (coverages
                .lazy()
                .join(taxonomy, on=COLUMN_GENOME_ID, how='left')
                .select(COLUMN_TAXONOMY,
                        pl.exclude(COLUMN_GENOME_ID, COLUMN_TAXONOMY)
    ).collect()
```

Reply:
Thanks! Integrated this to the next commit. `taxonomy`, which is loaded using `parse_taxonomy`, does not come in lazy (to be consistent with `parse_lengths`) therefore removing lazy() and .collect()